### PR TITLE
[UIPO-17] Join us component

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/JoinUs/JoinUs.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value }, () => {
+      this.validateField(value);
+    });
+  }
+
+  validateField(value) {
+    const emailRegEx = /^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$/;
+    const fieldValidationError = this.state.formError;
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.js]: Please use destructuring objects for state values

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

File: src/components/JoinUs/JoinUs.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ class JoinUs extends Component {
 
   validateField(value) {
     const emailRegEx = /^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$/;
-    const fieldValidationError = this.state.formError;
+    const { fieldValidationError } = this.state.formError;
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.js]: it should be
`const { formError, emailValid } = this.state;`
or
`const { formError: fieldValidationError, emailValid } = this.state;`

File: src/components/JoinUs/JoinUs.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  render() {
+    return (
+      <form className="join-us">
+        <label htmlFor="email" className="title">newsletter</label>
+        <div className="join-us-wrapper">
+          <div className="input-wrapper">
+            <input type="email" required name="email"
+              value={this.state.email}
+              onChange={(e)=>this.handleInput(e)}
+              className={`email-input ${this.errorClass(this.state.formError.email)}`} />
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.js]: Please use `className` library for add classes -> `cx({ warning: true })`

File: src/components/DefaultLayout/index.js [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import Header from '../Header';
+import Footer from '../Footer';
+
+const DefaultLayout = ({ component: Component, hideFooter, hideHeader, ...rest }) => {
+  return (
+    <Route {...rest} render={(matchProps) => (
+      <div>
```
Comment (AlyonaPianykh) [inline src/components/DefaultLayout/index.js]: extra div, should be deleted

File: src/components/JoinUs/JoinUs.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    this.setState({ email: value }, () => {
+      this.validateField(value);
+    });
+  }
+
+  validateField(value) {
+    const isEmailValid = value.match(emailRegEx);
+    this.setState({ isEmailValid }, () => {
+    });
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.js]: please remove callback function

File: src/components/JoinUs/JoinUs.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        <label htmlFor="email" className="title">newsletter</label>
+        <div className="join-us-wrapper">
+          <div className="input-wrapper">
+            <input type="email" required name="email"
+              value={email}
+              onChange={this.handleInput}
+              className={cx('email-input', { warning: !isEmailValid }, { success: isEmailValid })}
+            />
+            <button className="btn" type="submit" disabled={formValid}>
+              <i className="chevron right icon"></i>
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.js]: The empty tag should be closed by self

File: src/components/DefaultLayout/index.js (hunk 4 above)
Comment (Nazar-bnm) [reply src/components/DefaultLayout/index.js]: you can just use `<> </>`

File: src/components/JoinUs/JoinUs.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      <form className="join-us">
+        <label htmlFor="email" className="title">newsletter</label>
+        <div className="join-us-wrapper">
+          <div className="input-wrapper">
+            <input type="email" required name="email"
+              value={email}
+              onChange={this.handleInput}
+              className={cx('email-input', { warning: !isEmailValid }, { success: isEmailValid })}
+            />
+            <button className="btn" type="submit" disabled={formValid}>
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.js]: `formValid` value is never assigned

File: src/components/JoinUs/JoinUs.scss [hunk 8]
```diff
@@ -0,0 +1,52 @@
+@import '../../scss/global_styles';
+
+.join-us {
+    @include flex;
+}
+
+.title {
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.scss]: please use bem methodology

File: src/components/JoinUs/JoinUs.scss [hunk 9]
```diff
@@ -0,0 +1,52 @@
+@import '../../scss/global_styles';
+
+.join-us {
+    @include flex;
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.scss]: you have to use 2 spaces

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

File: src/components/JoinUs/JoinUs.scss [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  
+  &__title {
+    @include title-font;
+    @include flex($align: center);
+    color: $white;
+    font-size: 0.8rem;
+  }
+}
+
+.input-wrapper {
```
Comment (Nazar-bnm) [inline src/components/JoinUs/JoinUs.scss]: please use bem methodology everywhere!

**Review by Nazar-bnm — CHANGES_REQUESTED** [review]

**Comment by Hollister009** [comment]
I recommend opening new PR because this one contains too many changes

**Comment by ghost** [comment]
> Resolve conflicts!
> I recomend to open new PR, because this one contains too many changes

it means - hard-working)

**Comment by Hollister009** [comment]
> it means - hard-working)

Well not really in case of this PR...

File: src/components/DefaultLayout/index.js [hunk 11]
```diff
@@ -1,3 +1,27 @@
...
+import { Route } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import Header from '../Header';
+import Footer from '../Footer';
+
+const DefaultLayout = ({ component: Component, hideFooter, hideHeader, ...rest }) => {
+  return (
+    <Route {...rest} render={(matchProps) => (
+      <>
```
Comment (AlyonaPianykh) [inline src/components/DefaultLayout/index.js]: I don't know what `<>` mean but shouldn't be here at all =)

Comment (AlyonaPianykh) [reply src/components/DefaultLayout/index.js]: Please Check React.Fragment

**Comment by ghost** [comment]
Closed on request from @Hollister009

**Pull request closed without merging**